test/sjcl-ecdsa-recoverablepublickey-test.js

```diff
@@ -43,7 +43,7 @@ describe('ECDSA signing with recoverable public key', function(){
 
         var pub_val_point = secret_key._curve.G.mult(secret_key._exponent);
         var public_key = new sjcl.ecc.ecdsa.publicKey(curve, pub_val_point);
-        var hash = hash_function(message);
+        var hash = sjcl.bitArray.clamp(hash_function(message), secret_key._curveBitLength);
```

Comment:
See `sjcl-ecdsa-recoverablepublickey-test.js`:
```
        var hash = sjcl.bitArray.clamp(hash_function(message), secret_key._curveBitLength);

        var recoverable_signature = secret_key.signWithRecoverablePublicKey(hash, 0, random_value);
        var recovered_public_key = sjcl.ecc.ecdsa.publicKey.recoverFromSignature(hash, recoverable_signature);
```

The change looks like it was done because it's also done in the sjcl version of sign which we now use: 
```
  sign: function(hash, paranoia, fakeLegacyVersion, fixedKForTesting) {
    if (sjcl.bitArray.bitLength(hash) > this._curveBitLength) {
      hash = sjcl.bitArray.clamp(hash, this._curveBitLength);
    }
```

I'm not sure why the tests were using sha512 before, or what the ramifications of using it without clamping are ? But without it, the tests which use sha512 will fail.